ec2/auto_scaling_group/templates/scripts/asg_user_data.sh

```
@@ -0,0 +1,393 @@
+
+# Mackerel installation script details
+MACKEREL_SCRIPT_URL="https://mackerel.io/file/script/amznlinux/setup-all-yum-v2.sh"
+# Expected SHA-256 checksum (should be updated when script changes)
+# NOTE: To enable checksum verification, replace the placeholder with the actual SHA-256 checksum
+# You can get the current checksum by running: curl -fsSL https://mackerel.io/file/script/amznlinux/setup-all-yum-v2.sh | sha256sum
+# When set to placeholder value, verification will be skipped to prevent ASG startup failures
+EXPECTED_SHA256="placeholder_checksum_needs_verification"
+
+# Create temporary file for script download
+TEMP_SCRIPT=$(mktemp /tmp/mackerel-install.XXXXXX)
+trap "rm -f $TEMP_SCRIPT" EXIT
```

Shellcheck (0.10.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)